transport/tchannel/peer.go

```diff
@@ -165,6 +166,17 @@ func (p *tchannelPeer) sleep(delay time.Duration) (completed bool) {
 	return false
 }
 
+func (p *tchannelPeer) getPeer() *tchannel.Peer {
+	return p.getRootPeers().GetOrAdd(p.HostPort())
+}
+
+func (p *tchannelPeer) getRootPeers() *tchannel.RootPeerList {
+	if p.transport.ch == nil {
```

Comment:
When you compare this method with deleted `peerList()` in `transport.go`, `ch` is being accessed without locks. It is safe:
* `tchannelPeer.maintainConnection()` - this method invokes `getRootPeers()` after waiting for transport to start with `<-p.transport.once.Started()` (unblocks once transport has started where it creates `ch`)
* `tchannelPeer.Call()`- this method invokes `getRootPeers()` once transport has started.